# Assignment 2: Estimation And Confidence Intervals

A. Build 99% Confidence Interval Using Sample Standard Deviation
Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.


In [3]:
import numpy as np
import pandas as pd
import statistics as st

In [99]:
data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])

In [105]:
# Calculation degrees of freedom = (total number of elements - 1)

dof = len(x) - 1
dof

14

In [109]:
#Calculating Mean of the data
mu = np.mean(data)
mu

1.2386666666666666

In [121]:
#Calculating Sample Standard Deviation
std_x = np.std(x, ddof = 1)    # here we must explicitly set ddof = 1 since we are calculating sample SD
std_x

0.19316412956959936

In [ ]:
# The reason why we are using t-Distribution is because our sample size is 15 which is lessthan 30 we'll be using t-distribution.

In [129]:
from scipy import stats

confidence_score = 0.99    #Confidence score 99% (given)
t_stats = stats.t.ppf(1 - (1-confidence_score)/2, dof)
t_stats

2.976842734370834

In [125]:
#Calculating margine of error

margine_of_error = t_value * (std_x/np.sqrt(len(x)))
margine_of_error

0.14846932822817596

In [136]:
lower_bound = mu - margine_of_error
upper_bound = mu + margine_of_error
print("The Confidence Interval : ")
print("Lower Bound : ",lower_bound)
print("Upper Bound : ",upper_bound)

The Confidence Interval : 
Lower Bound :  1.0901973384384906
Upper Bound :  1.3871359948948425


------------------------------------------------------------------------------

B. Build 99% Confidence Interval Using Known Population Standard Deviation
If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.


In [145]:
#calculate the sample mean

mu = data.mean()   #sample mean
dof = len(data)-1  #degrees of freedom
population_sd = 0.2   #population standard devation (given in the question)
confidence_score = 0.99   #confidence score meaning (alpha = 0.01)
alpha = 1 - confidence_score

In [147]:
# Since we are already aware of population standard deviation our data points are definately more than 30 so we'll have to use z-test

In [153]:
z_value = stats.norm.ppf(1-(1-confidence_score)/2)
z_value

2.5758293035489004

In [155]:
margine_of_error = z_value * (population_sd/np.sqrt(len(data)))
margine_of_error

0.13301525327090588

In [157]:
lower_bound = mu - margine_of_error
upper_bound = mu + margine_of_error
print("The Confidence Interval : ")
print("Lower Bound : ",lower_bound)
print("Upper Bound : ",upper_bound)

The Confidence Interval : 
Lower Bound :  1.1056514133957607
Upper Bound :  1.3716819199375725


From above 2 Confidence intervals:
t-Distribution Confidence interval : (1.0901973384384906, 1.3871359948948425)
z-Distribution Confidence interval : (1.1056514133957607, 1.3716819199375725)

We can infer that :
* z-Distribution range is smaller than t-Distribution.
* z-Distribution is more precise than t-Distribution.
* If we don't know the population standard deviation we use t-Distribution.